### PR TITLE
Move Roslyn to latest 15.3 version

### PR DESCRIPTION
**Pull request by ellismg**
Move to the latest 15.3 versions of Roslyn.  The previous one we had was over five months old.

### CODE REVIEW

**Comment by ellismg** [comment]
Looking into the crossgen failure...

**Comment by ellismg** [comment]
@dotnet-bot test Linux x64 Release Build please

**Comment by ellismg** [comment]
@eerhardt Can you take a look?  The package has renamed csc.exe and vbc.exe to csc.dll and vbc.dll.  I tried to keep that rename (adjusting the Run[CSC]VBC] scripts) but I was hitting problem later in a test still looking for csc.exe and I don't really have cycles to figure out what was going on there.

/cc @jaredpar (can you ensure I picked up a sensible version)?

**Comment by ellismg** [comment]
@MattGertz This is bringing a new version of Roslyn into the CLI (the previous version was quite old).  Are you OK with this happening?

**Comment by MattGertz** [comment]
@livarcocc @jaredpar Can you folks think of any reason why this would be a problem?  Seems a normal good thing to do, assuming validation has been done.

File: src/tool_roslyn/tool_roslyn.csproj [hunk 1]
```diff
@@ -16,7 +16,7 @@
     <PackageReference Include="Microsoft.CodeAnalysis.CSharp" Version="$(CLI_Roslyn_Version)" />
     <PackageReference Include="Microsoft.CodeAnalysis.Build.Tasks" Version="$(CLI_Roslyn_Version)" />
     <PackageReference Include="Microsoft.Net.Compilers.netcore" Version="$(CLI_Roslyn_Version)" />
-    <PackageReference Include="Microsoft.DiaSymReader.Native" Version="1.4.1" />
+    <PackageReference Include="Microsoft.DiaSymReader.Native" Version="1.6.0-beta2-25304" />
```
Comment (eerhardt) [inline src/tool_roslyn/tool_roslyn.csproj]: Can you move this to DependencyVersions.props?

File: src/dotnet/dotnet.csproj [hunk 2]
```diff
@@ -32,6 +32,10 @@
     <PackageReference Include="System.Diagnostics.TextWriterTraceListener" Version="4.3.0" />
     <PackageReference Include="System.Runtime.Serialization.Primitives" Version="4.3.0" />
     <PackageReference Include="System.Private.DataContractSerialization" Version="4.3.0" />
+    <!-- The project json migration commands depend on an older version of Roslyn.
+         Lift the version here to match what tool_roslyn depends on (otherwise an older version will
+         be added to the TPA when we crossgen and we won't be able to crossgen tool_roslyn -->
+    <PackageReference Include="Microsoft.CodeAnalysis.CSharp" Version="$(CLI_Roslyn_Version)" />
```
Comment (eerhardt) [inline src/dotnet/dotnet.csproj]: Something like this more belongs in `src\redist\redist.csproj`.  That's the real project that aggregates `dotnet`, `msbuild`, and `nuget` into the official output in the `sdk` folder.  You want something like this as high as possible so it always wins over some dependency's version.

Can you move this to redist.csproj instead?  This probably wouldn't cause issues today, but down the line if someone new starts referencing Roslyn too.

File: src/tool_roslyn/tool_roslyn.csproj (hunk 1 above)
Comment (ellismg) [inline src/tool_roslyn/tool_roslyn.csproj]: Of course.

File: src/dotnet/dotnet.csproj (hunk 2 above)
Comment (ellismg) [inline src/dotnet/dotnet.csproj]: Sure, I'll give that a whirl.

File: build/DependencyVersions.props [hunk 3]
```diff
@@ -3,7 +3,8 @@
   <PropertyGroup>
     <CLI_SharedFrameworkVersion>2.0.0-preview2-25319-02</CLI_SharedFrameworkVersion>
     <CLI_MSBuild_Version>15.3.0-preview-000246-05</CLI_MSBuild_Version>
-    <CLI_Roslyn_Version>2.0.0-rc4-61325-08</CLI_Roslyn_Version>
+    <CLI_Roslyn_Version>2.3.0-dev-61716-04</CLI_Roslyn_Version>
```
Comment (jaredpar) [inline build/DependencyVersions.props]: Please use 2.3.0-beta2-61716-09

Comment (ellismg) [inline build/DependencyVersions.props]: Understood.

**Comment by ellismg** [comment]
@livarcocc I'm using the version that Jared recommended and I responded to @eerhardt's feedback.  From my end, this is ready to go whenever you want to merge it (my reading of Getz's comment is that if you're okay with this then it can go in).

Please merge if you are comfortable with this or let me know what else I can do to move this along...

**Comment by livarcocc** [comment]
@mattgertz it seemed like you approved already. Merged it.

**Pull request merged**